codec/src/main/java/io/netty/handler/codec/compression/Lz4XXHash32.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.compression;
+
+import io.netty.buffer.ByteBuf;
+import net.jpountz.xxhash.StreamingXXHash32;
+import net.jpountz.xxhash.XXHash32;
+import net.jpountz.xxhash.XXHashFactory;
+
+import java.nio.ByteBuffer;
+import java.util.zip.Checksum;
+
+/**
+ * A special-purpose {@link ByteBufChecksum} implementation for use with
+ * {@link Lz4FrameEncoder} and {@link Lz4FrameDecoder}.
+ *
+ * {@link StreamingXXHash32#asChecksum()} has a particularly nasty implementation
+ * of {@link Checksum#update(int)} that allocates a single-element byte array for
+ * every invocation.
+ *
+ * In addition to that, it doesn't implement an overload that accepts a {@link ByteBuffer}
+ * as an argument.
+ *
+ * Combined, this means that we can't use {@code ReflectiveByteBufChecksum} at all,
+ * and can't use {@code SlowByteBufChecksum} because of its atrocious performance
+ * with direct byte buffers (allocating an array and making a JNI call for every byte
+ * checksummed might be considered sub-optimal by some).
+ *
+ * Block version of xxHash32 ({@link XXHash32}), however, does provide
+ * {@link XXHash32#hash(ByteBuffer, int)} method that is efficient and does exactly
+ * what we need, with a caveat that we can only invoke it once before having to reset.
+ * This, however, is fine for our purposes, given the way we use it in
+ * {@link Lz4FrameEncoder} and {@link Lz4FrameDecoder}:
+ * {@code reset()}, followed by one {@code update()}, followed by {@code getValue()}.
+ */
+public class Lz4XXHash32 extends ByteBufChecksum {
```

Comment:
`final class` ? (package-private + final) ?

Reply:
`final` can do, but it should stay `public` - I know of at least one codebase that uses xxhash but overrides the seed. Need a way for those users to keep using xxhash with their own seed and decent perf.

One other way would be to allow new `Lz4FrameEncoder`/`Lz4FrameDecoder` constructors, but that felt very unclean.

Reply:
Addressed `final` in https://github.com/netty/netty/pull/9249/commits/c71a83f6f085c6f3bab70c60fe8c6a29ba54d517

Reply:
Then lets use final :)

Reply:
Messages raced (: